### PR TITLE
[FIX] stock_account: _run_fifo_vacuum

### PR DESCRIPTION
**Pull request by sle-odoo**
If the oldest problematic layer cannot be fixed, no need to try to fix
later one.

### CODE REVIEW

File: addons/stock_account/models/product.py [hunk 1]
```diff
@@ -331,7 +331,7 @@ def _run_fifo_vacuum(self, company=None):
             ]
             candidates = self.env['stock.valuation.layer'].sudo().search(domain)
             if not candidates:
-                continue
+                break
```
Comment (pimodoo) [inline addons/stock_account/models/product.py]: put reference of OPW in commit

File: addons/stock_account/models/stock_valuation_layer.py [hunk 2]
```diff
@@ -31,3 +31,9 @@ class StockValuationLayer(models.Model):
     stock_move_id = fields.Many2one('stock.move', 'Stock Move', readonly=True, check_company=True)
     account_move_id = fields.Many2one('account.move', 'Journal Entry', readonly=True, check_company=True)
 
+    def init(self):
```
Comment (pimodoo) [inline addons/stock_account/models/stock_valuation_layer.py]: no REF in stable rename FIX

File: addons/stock_account/models/product.py [hunk 3]
```diff
@@ -317,7 +317,7 @@ def _run_fifo_vacuum(self, company=None):
             ('remaining_qty', '<', 0),
             ('stock_move_id', '!=', False),
             ('company_id', '=', company.id),
-        ])
+        ], order='create_date, id')
```
Comment (pimodoo) [inline addons/stock_account/models/product.py]: useless, already in model

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by nim-odoo** [comment]
robodoo r+

**Comment by nim-odoo** [comment]
robodoo retry

**Comment by nim-odoo** [comment]
@xmo-odoo runbot is green, but no CI tag?

**Comment by xmo-odoo** [comment]
@nim-odoo 1530 or so is about when we migrated the mergebot to 13.0, so it's probable that a few CI+ or whatever got lost at that point.

**Comment by nim-odoo** [comment]
robodoo r+

**Pull request closed without merging**